src/AutoRest.CSharp/Mgmt/AutoRest/MgmtConfiguration.cs

```diff
@@ -13,76 +13,119 @@ public class MgmtConfiguration
         {
-            OperationGroupToResourceType = operationGroupToResourceType?.ValueKind == JsonValueKind.Null ? new Dictionary<string, string>() : JsonSerializer.Deserialize<Dictionary<string, string>>(operationGroupToResourceType.ToString());
-            OperationGroupToResource = operationGroupToResource?.ValueKind == JsonValueKind.Null ? new Dictionary<string, string>() : JsonSerializer.Deserialize<Dictionary<string, string>>(operationGroupToResource.ToString());
-            OperationGroupToParent = operationGroupToParent?.ValueKind == JsonValueKind.Null ? new Dictionary<string, string>() : JsonSerializer.Deserialize<Dictionary<string, string>>(operationGroupToParent.ToString());
+            OperationGroupToResourceType = !IsValidJsonElement(operationGroupToResourceType) ? new Dictionary<string, string>() : JsonSerializer.Deserialize<Dictionary<string, string>>(operationGroupToResourceType.ToString());
+            OperationGroupToResource = !IsValidJsonElement(operationGroupToResource) ? new Dictionary<string, string>() : JsonSerializer.Deserialize<Dictionary<string, string>>(operationGroupToResource.ToString());
+            OperationGroupToParent = !IsValidJsonElement(operationGroupToParent) ? new Dictionary<string, string>() : JsonSerializer.Deserialize<Dictionary<string, string>>(operationGroupToParent.ToString());
+            SingletonResource = singletonResource;
             OperationGroupIsTuple = operationGroupIsTuple;
         }
 
         public IReadOnlyDictionary<string, string> OperationGroupToResourceType { get; }
         public IReadOnlyDictionary<string, string> OperationGroupToResource { get; }
         public IReadOnlyDictionary<string, string> OperationGroupToParent { get; }
+        public string[] SingletonResource { get; }
         public string[] OperationGroupIsTuple { get; }
 
         internal static MgmtConfiguration GetConfiguration(IPluginCommunication autoRest)
         {
             return new MgmtConfiguration(
                 autoRest.GetValue<string[]?>("operation-group-is-tuple").GetAwaiter().GetResult() ?? Array.Empty<string>(),
+                autoRest.GetValue<string[]?>("singleton-resource").GetAwaiter().GetResult() ?? Array.Empty<string>(),
                 autoRest.GetValue<JsonElement?>("operation-group-to-resource-type").GetAwaiter().GetResult(),
                 autoRest.GetValue<JsonElement?>("operation-group-to-resource").GetAwaiter().GetResult(),
-                autoRest.GetValue<JsonElement?>("operation-group-to-parent").GetAwaiter().GetResult()
-            );
+                autoRest.GetValue<JsonElement?>("operation-group-to-parent").GetAwaiter().GetResult());
         }
 
         internal void SaveConfiguration(Utf8JsonWriter writer)
         {
-            writer.WriteStartObject(nameof(OperationGroupToResourceType));
-            foreach (var keyval in OperationGroupToResourceType)
+            if (OperationGroupToResourceType.Count > 0)
```

Comment:
This pattern of:

```
if (things > 0)
{
    writer.WriteStartObject(nameof(thing));
    foreach (var keyval in things)
    {   
        writer.WriteString(keyval.Key, keyval.Value);
   }
   writer.WriteEndObject();
}
```

happens a number of times here, maybe extra into a function or extension?